### PR TITLE
built-in fonts and custom fonts examples

### PR DESCRIPTION
**Pull request by alexvoss**
Pull request for fonts examples
https://github.com/mkdocs-material/internal/issues/29

### CODE REVIEW

File: examples/fonts-builtin/docs/README.md
Comment (squidfunk) [inline examples/fonts-builtin/docs/README.md:28]: I'd say we should mention the privacy plugin to have a "best of both worlds" solution ☺️ Not sure where, but a small admonition that explains the plugin (just take one from our docs) might be good

**Review by squidfunk — CHANGES_REQUESTED** [review]

File: examples/fonts-builtin/docs/README.md
Comment (alexvoss) [reply examples/fonts-builtin/docs/README.md:28]: Add the tip - doing the same in the custom fonts example I just pushed a draft of.

**Comment by alexvoss** [comment]
I am trying to find a CDN for Noto (or another font really) that is not Google and that is not like Google. I have put in cdnfonts.com for now but I can't seem to find much info on who they are, how they make their money, etc. Do you know anything about jsDeliver? That seems to be the default one for Noto.

**Comment by alexvoss** [comment]
@squidfunk I will need to go for a walk (rare sunshine, so will head to the beach) and do a little shopping. Any feedback on this very welcome. I hope I can finish off any remaining work in the afternoon.

File: examples/fonts-builtin/docs/README.md
Comment (squidfunk) [inline examples/fonts-builtin/docs/README.md]: Typo "not" -> "no"

File: examples/fonts-builtin/docs/README.md
Comment (squidfunk) [inline examples/fonts-builtin/docs/README.md]: Typo "specifc" -> "specify"

File: examples/fonts-custom/docs/README.md
Comment (squidfunk) [inline examples/fonts-custom/docs/README.md]: I'd just call this "alongside your documentatin" or something like it. "Server" might be off-putting, because it sounds like you need to set up hosting yourself. Just put them in docs, deploy to GitHub pages, done.

File: examples/fonts-custom/docs/README.md
Comment (squidfunk) [inline examples/fonts-custom/docs/README.md]: See above

File: examples/fonts-custom/docs/README.md
Comment (squidfunk) [inline examples/fonts-custom/docs/README.md:103]: Why not use Snippets here? Less duplication and the code example will always be up to date ☺️

**Review by squidfunk — CHANGES_REQUESTED** [review]
I'm currently busy working on the search, so I didn't run the PR, only minor changes. If it works and you feel comfortable deploying it, I'm fine with merging it and linking it in the docs.

File: examples/fonts-custom/docs/README.md
Comment (alexvoss) [reply examples/fonts-custom/docs/README.md:103]: Snippets need to be added to `mkdocs.yml` and so that is no longer minimal. Already had that with admonitions.

**Comment by alexvoss** [comment]
Right, should be fine now. Fingers crossed. I did test that both examples work with both insider and public version.

File: examples/fonts-custom/docs/README.md
Comment (squidfunk) [inline examples/fonts-custom/docs/README.md]: One final question: what line gutters are you using? I'm trying to limit everything to 80 chars, as it allows to render code blocks in PDFs nicely, but it looks like you're using 70 or something in that area. Could we agree on 80? Sorry to be so pendantic 😅

Comment (alexvoss) [reply examples/fonts-custom/docs/README.md]: Indeed, it seems that for some reason it is 70 on `.md` files. This is `vim`'s doing. I have changed it to 80. Shame this does not seem to be something that a `.editorconfig` could take care of?! Do you want me to change these or just use 80 from here onwards?

Comment (alexvoss) [reply examples/fonts-custom/docs/README.md]: I am also just realizing that I do not understand the comment about code blocks. Those sometimes get longer, especially when URLs are involved. I am not sure what to do about that.

You can stop apologizing for being "pedantic" - that is quite ok and we know each other well enough by now!

Comment (squidfunk) [reply examples/fonts-custom/docs/README.md]: > Indeed, it seems that for some reason it is 70 on .md files. This is vim's doing. I have changed it to 80. Shame this does not seem to be something that a .editorconfig could take care of?! Do you want me to change these or just use 80 from here onwards?

Please just use 80 onwards. When we revisit those docs at some point, we can correct them the next time.

> I am also just realizing that I do not understand the comment about code blocks. Those sometimes get longer, especially when URLs are involved. I am not sure what to do about that.

When you print a site made with Material for MkDocs from the browser, limiting code blocks to 80 characters will always nicely fit the page. That's one of the reasons I always stick to the 80 characters rule. Additionally, it allows to open multiple editor tabs next to each other without having to scroll horizontally. I'm just not that big a fan of horizontal scrolling on text ☺️ So to sum up: 80 is just the best practice I've been following for many years.

> You can stop apologizing for being "pedantic" - that is quite ok and we know each other well enough by now!

Ack!

Comment (alexvoss) [reply examples/fonts-custom/docs/README.md]: I don't question the 80 chars rule (I had 70 configured, probably on the basis of recommended line length for reading), just wondering what to do with things like this in the CSS file:

```
https://cdn.jsdelivr.net/gh/notofonts/notofonts.github.io/fonts/NotoSerif/unhinted/otf/NotoSerif-Bold.otf
```

**Comment by alexvoss** [comment]
Not sure where we are with this - I thought this had runs its course but GitHub says there are change request to be addressed? I will ask you to review again, mainly because I don't know what else to do.

**Comment by alexvoss** [comment]
@squidfunk one example that is missing is that of auto-loading another font from Google Fonts. That does raise the question of data protection and whether we should not use the privacy plugin here. I do note that Google Analytics is turned on, so would we not also need to get peoples' consent?

**Comment by squidfunk** [comment]
Hey, sorry that I'm sometimes unresponsive right now, I'm focusing on the search, so I try to minimize distractions, which is why some things slip.

> That does raise the question of data protection and whether we should not use the privacy plugin here.

We can use the plugin, but that would make the example impossible to run without Insiders. Note that Google Analytics is configured in `mkdocs.yml`, but no `GOOGLE_ANALYTICS_KEY` variable is set during the build, so the integration is not active. It's only Google Fonts that is loaded.

I'd say we keep it simple.

__Edit__: you're also [using CDNs for the non-local example](https://github.com/mkdocs-material/examples/pull/11/files#diff-d666f757ba97a9d4483a63a17e77fa725eaffe23984826f4a8f012dd3b5283fb), which would mandate for the same logic. I take the risk and get sued over using a CDN before getting consent for every third-party request in any example. Adding the cookie consent to each of them defies the meaning of examples.

File: examples/fonts-builtin/mkdocs.yml
Comment (squidfunk) [inline examples/fonts-builtin/mkdocs.yml]: Typo `broswer` 😉 Did you check for typos?

File: examples/fonts-custom/docs/assets/stylesheets/extra.css
Comment (squidfunk) [inline examples/fonts-custom/docs/assets/stylesheets/extra.css]: Please break those onto separate lines (other CSS files as well)

File: examples/fonts-custom/docs/README.md
Comment (squidfunk) [inline examples/fonts-custom/docs/README.md]: I'm not sure we need this example – it's actually only the first step for self hosting, as the font files are still hosted on the CDN. I think I would drop this one and recommend the privacy plugin which takes care of all of this. With the privacy plugin, you get self hosting with the first example and a single line of config. It will be released when we reach the next funding goal, which is just around the corner.

**Review by squidfunk — CHANGES_REQUESTED** [review]

File: examples/fonts-builtin/mkdocs.yml
Comment (alexvoss) [reply examples/fonts-builtin/mkdocs.yml]: another config change for my `.vimrc`, so should be better in the future.

**Comment by alexvoss** [comment]
> We can use the plugin, but that would make the example impossible to run without Insiders.

We could use the privacy plugin for building the whole set of examples and apply this to the individual examples through the transformation script. That would mean the deployed version would be GDPR compliant and the downloadable examples would not assume the use of Insiders.

**Comment by alexvoss** [comment]
Ok, I think I have addressed the requested changes. Also moved the mention of the privacy plugin closer to the top of the page on the custom fonts example.

**Review by squidfunk — APPROVED** [review]

**Comment by squidfunk** [comment]
LGTM! By the way, once a PR is approved on this repository, feel free to merge it whenever you feel it's ready. On the actual Material for MkDocs repository, we need to carefully weigh when PRs can be merged, because it impacts a lot of users, but this repository is for examples/demonstration only, so we can relax it a bit.

**Pull request merged**
